app/services/legacy/dataset_import_service.rb

```diff
@@ -35,49 +35,85 @@ def update_or_create_dataset
+    if @legacy_dataset.has_key?('additional_resources')
+      @legacy_dataset['additional_resources'].each do |resource|
+        datafile = Doc.find_or_create_by(url: resource["url"], dataset_id: dataset.id)
+        base_attributes = create_datafile_base_attributes(resource, dataset)
+
+        datafile.assign_attributes(base_attributes)
+        datafile.save!(validate: false)
+      end
     end
   end
 
-  def create_resource(resource, dataset)
-    datafile = find_or_initialize(resource)
-    datafile.uuid = resource["id"]
-    datafile.format = resource["format"]
-    datafile.name = resource["description"].presence || "No name specified"
-    datafile.created_at = dataset.created_at
-    datafile.updated_at = dataset.last_updated_at
-
-    if resource["date"].present? && !documentation?(resource['format'])
-      dates = get_start_end_date(resource["date"])
-      if dataset.frequency != 'never'
-        begin
-          datafile.start_date = Date.parse(dates[0])
-        rescue ArgumentError
-          datafile.start_date = Date.new(1,1,1)
-        end
-        begin
-          datafile.end_date = Date.parse(dates[1])
-        rescue ArgumentError
-          datafile.end_date = Date.new(1,1,1)
-        end
+  def create_non_timeseries_datafiles(dataset)
+    if @legacy_dataset.has_key?('individual_resources')
+      @legacy_dataset['individual_resources'].each do |resource|
+        datafile = Doc.find_or_create_by(url: resource["url"], dataset_id: dataset.id)
+        base_attributes = create_datafile_base_attributes(resource, dataset)
+
+        datafile.assign_attributes(base_attributes)
+        datafile.save!(validate: false)
       end
     end
+  end
+
+  def create_timeseries_datafiles(dataset)
+    if @legacy_dataset.has_key?('timeseries_resources')
+      @legacy_dataset['timeseries_resources'].each do |resource|
+        datafile = Link.find_or_create_by(url: resource["url"], dataset_id: dataset.id)
+        base_attributes = create_datafile_base_attributes(resource, dataset)
+        date_attributes = create_datafile_date_attributes(resource)
 
-    datafile.day = datafile.end_date.day
-    datafile.month = datafile.end_date.month
-    datafile.year = datafile.end_date.year
+        datafile.assign_attributes(base_attributes)
+        datafile.assign_attributes(date_attributes)
 
-    datafile.save!(validate: false)
+        datafile.save!(validate: false)
+      end
+    end
+  end
+
+  def create_datafile_base_attributes(resource, dataset)
+    {
+      uuid: resource["id"],
+      format: resource["format"],
+      name: datafile_name(resource),
+      created_at: dataset.created_at,
+      updated_at: dataset.last_updated_at
+    }
+  end
+
+  def create_datafile_date_attributes(resource)
+    dates = get_start_end_date(resource["date"])
+    start_date = Date.parse(dates[0])
+    end_date = Date.parse(dates[1])
+
+    {
+      start_date: start_date,
+      end_date: end_date,
```

Comment:
this makes me think we should have links and docs set the dates the same way, i.e. using `day` `month`, and `year`.

Reply:
something for another PR.